packages/server-core/src/integrations/zendesk/zendesk.class.ts

```diff
@@ -0,0 +1,53 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { ServiceInterface } from '@feathersjs/feathers/lib/declarations'
+
+import { Application } from '../../../declarations'
+
+import { sign } from 'jsonwebtoken'
+
+import { ZendeskAuthenticationType } from '@etherealengine/common/src/schema.type.module'
+import appConfig from '@etherealengine/server-core/src/appconfig'
+
+/**
+ * A class for Signing zendesk tokens service
+ */
+export class ZendeskAuthenticationService implements ServiceInterface<string, ZendeskAuthenticationType> {
+  app: Application
+
+  constructor(app: Application) {
+    this.app = app
+  }
+
+  async create(data: ZendeskAuthenticationType): Promise<string> {
```

Comment:
You can do this in a hook.

`packages/projects/projects/eepro-multitenancy/services/entri/entri-auth` is a similar example of who its uses hooks.

The .class file should inherit from `BaseService` like in `entri-auth.class.ts`

Reply:
Done! Moved logic from service to hooks